### PR TITLE
Fixed size of iframe on modal opening

### PR DESCRIPTION
**Pull request by Wassaf-Shahzad**
#### What are the relevant tickets?
Fixes the LMS issue for ticket #308 

#### What's this PR do?
This PR updated the iframe size on modal opening and closing

#### How should this be manually tested?
This  PR can be manually tested by following steps
    - Create a SGA unit in Studio
    - Open the unit on LMS with a staff user
    - Press Grade submissions button and verify popup is not cutoff

#### Screenshots (if appropriate)
(Optional)

Screenshot 2021-12-22 at 5 50 03 PM

Screenshot 2021-12-22 at 5 49 57 PM

### CODE REVIEW

File: edx_sga/static/js/src/edx_sga.js [hunk 1]
```diff
@@ -330,6 +331,8 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                 block.find('#grade-submissions-button')
                     .leanModal()
                     .on('click', function() {
+                        sendResizeMessage(document.body.clientHeight * 2)
```
Comment (umarmughal824) [inline edx_sga/static/js/src/edx_sga.js:334]: I have tried it but when I keep the page content that spans to two pages. So, when I open up that popup it spans to four pages that doesn't make sense at all

I have attached a picture for your reference
[screencapture-localhost-2000-course-course-v1-edX-DemoX-Demo-Course-block-v1-edX-DemoX-Demo-Course-type-sequential-block-basic-questions-block-v1-edX-DemoX-Demo-Course-type-vertical-block-2152d4a4aadc4cb0af5256394a3d1fc7-2021-12-28-18_40_57.pdf](https://github.com/mitodl/edx-sga/files/7784930/screencapture-localhost-2000-course-course-v1-edX-DemoX-Demo-Course-block-v1-edX-DemoX-Demo-Course-type-sequential-block-basic-questions-block-v1-edX-DemoX-Demo-Course-type-vertical-block-2152d4a4aadc4cb0af5256394a3d1fc7-2021-12-28-18_40_57.pdf)

[screencapture-localhost-2000-course-course-v1-edX-DemoX-Demo-Course-block-v1-edX-DemoX-Demo-Course-type-sequential-block-basic-questions-block-v1-edX-DemoX-Demo-Course-type-vertical-block-2152d4a4aadc4cb0af5256394a3d1fc7-2021-12-28-18_41_38.pdf](https://github.com/mitodl/edx-sga/files/7784932/screencapture-localhost-2000-course-course-v1-edX-DemoX-Demo-Course-block-v1-edX-DemoX-Demo-Course-type-sequential-block-basic-questions-block-v1-edX-DemoX-Demo-Course-type-vertical-block-2152d4a4aadc4cb0af5256394a3d1fc7-2021-12-28-18_41_38.pdf)

Comment (umarmughal824) [reply edx_sga/static/js/src/edx_sga.js:334]: On small pages it looks like that is also very weird

Screenshot 2021-12-28 at 18 10 58
Screenshot 2021-12-28 at 18 44 54
Screenshot 2021-12-28 at 18 45 01
Screenshot 2021-12-28 at 18 45 09

**Review by umarmughal824 — CHANGES_REQUESTED** [review]
Please see the comments

Also in my opinion that should be ideal to resolve through CSS where we would like to keep the page size something the total screen size.

Second, it does not even look like the legacy view. Please take a look at the lega view by the clicking on the Legacy View in the navbar of the section so that you could get the idea of it how it should look like

File: edx_sga/static/js/src/edx_sga.js [hunk 2]
```diff
@@ -376,8 +381,47 @@ function StaffGradedAssignmentXBlock(runtime, element) {
...
+            }
+            })
         });
 
+        function updateIframe(){
+          if(isInIframe && !currentIFrameHeight){
+            currentIFrameHeight = $("body").height()
+            addMaxHeightInIframe()
+            if(currentIFrameHeight < 600){
+              sendResizeMessage(currentIFrameHeight + 600)
```
Comment (umarmughal824) [inline edx_sga/static/js/src/edx_sga.js:397]: I am thinking what if the height of the body is already closed to 600 but not exactly 600 so adding another 600 would not be ideal. In fact, that would add another scroll and the iframe pop size would again become more than the screen size. What's your opinion on that. Could you try to check the behavior of that manually in inspect mode?

File: edx_sga/static/js/src/edx_sga.js [hunk 3]
```diff
@@ -376,8 +381,47 @@ function StaffGradedAssignmentXBlock(runtime, element) {
...
+            }
+          }
+        }
+
+        function isInIframe(){
+          return window.parent !== window
+        }
+
+        function addMaxHeightInIframe(){
+            $(".grade-submission").css("max-height", "600px")
```
Comment (umarmughal824) [inline edx_sga/static/js/src/edx_sga.js:407]: don't you think, it should rather be applied on iframe so that it could cover all the iframe size rather on a particular iframe with `.grade-submission`?

Like we have Staff debug, or other iframes too. So these other iframes would remain scattered and distorted. Let me know what's your opinion on that.

Comment (umarmughal824) [reply edx_sga/static/js/src/edx_sga.js:407]: And if we go to choose to apply it on `iframe`, it should be a part of `.scss/css`. Just try to load another popup like Staff Debug / Debug info with staff user

**Review by umarmughal824 — CHANGES_REQUESTED** [review]

File: edx_sga/static/js/src/edx_sga.js [hunk 4]
```diff
@@ -376,8 +381,47 @@ function StaffGradedAssignmentXBlock(runtime, element) {
...
+          // This blocks checks to see if the xBlock is part 
+          // of Learning MFE
+          if (window.parent !== window){
+            window.parent.postMessage({
+              type: 'plugin.resize',
+              payload: {
+                height: height,
+                }
+              }, document.referrer
+            );
```
[the remark this answers is not in the document]
Comment (pdpinch) [reply edx_sga/static/js/src/edx_sga.js:421]: should we add a code formatter to this repo?

[the remark this answers is not in the document]
Comment (Wassaf-Shahzad) [reply edx_sga/static/js/src/edx_sga.js]: fixed indentation for the whole of edx-sga.js.
We should probably add  formatter as well.

File: edx_sga/static/js/src/edx_sga.js (hunk 2 above)
Comment (Wassaf-Shahzad) [reply edx_sga/static/js/src/edx_sga.js]: This wont be a problem  since we are also adding a max-height constraint to fix the popup blowing up incase the iframe itself is too large

File: edx_sga/static/js/src/edx_sga.js (hunk 3 above)
Comment (Wassaf-Shahzad) [reply edx_sga/static/js/src/edx_sga.js]: The iframe tag is out of scope from the SGA block and so we cant access it using it using jQuery or vanilla JS.
We could however move the style changes from edx-sga repo to learner-MFE as that controls the various Iframes rendered here.

File: edx_sga/static/js/src/edx_sga.js (hunk 2 above)
Comment (umarmughal824) [reply edx_sga/static/js/src/edx_sga.js]: then why are we adding 600 to currentIframeheight? Instead, we should set it to 600.

Comment (umarmughal824) [inline edx_sga/static/js/src/edx_sga.js]:
```suggestion
              sendResizeMessage(600)
```

File: edx_sga/static/js/src/edx_sga.js (hunk 3 above)
Comment (umarmughal824) [reply edx_sga/static/js/src/edx_sga.js]: and does it have an effect the legacy view?

Comment (Wassaf-Shahzad) [reply edx_sga/static/js/src/edx_sga.js]: No

File: edx_sga/static/js/src/edx_sga.js (hunk 2 above)
Comment (Wassaf-Shahzad) [reply edx_sga/static/js/src/edx_sga.js]: done

File: edx_sga/static/css/edx_sga.css [hunk 5]
```diff
@@ -1,13 +1,16 @@
 /* CSS for StaffGradedAssignmentXBlock */
 
+.modal-iframe-height {
+    max-height: 600px;
+}
+
```
Comment (umarmughal824) [inline edx_sga/static/css/edx_sga.css:6]: I guess that is changing the legacy behavior, but not sure. You need to confirm it

**Review by umarmughal824 — CHANGES_REQUESTED** [review]
It's also changing the legacy behavior, at least we should not change it.

Screenshot 2022-01-25 at 16 43 23
Screenshot 2022-01-25 at 16 44 24

**Comment by pdpinch** [comment]
@Wassaf-Shahzad are you saying that we should upstream a change to https://github.com/openedx/frontend-app-learning ?

**Comment by Wassaf-Shahzad** [comment]
> @Wassaf-Shahzad are you saying that we should upstream a change to https://github.com/openedx/frontend-app-learning ?

This issue can be fixed by creating an upstream to the edx-platform not the learner MFE as I first thought as the issues exists on both sides (Legacy and Learner MFE) I also created a new issue to track this and other variants.
-  https://github.com/mitodl/edx-sga/issues/315

This issue an be reproduced in the QA environment  as explained in the above mentioned issue.
TLDR
The size of the content of the modal is smaller than the enforced `80%` height attribute which causes the extra black area to render.
